src/clr-angular/utils/drag-and-drop/droppable/droppable.ts

```diff
@@ -0,0 +1,205 @@
+import {ClrDragAndDropEventBus} from "../providers/drag-and-drop-event-bus";
+
+@Directive({
+    selector: "[clrDroppable]",
+    providers: [DomAdapter],
+    host: {class: "droppable", "[class.draggable-match]": "isDraggableMatch"}
+})
+export class ClrDroppable<T> implements OnInit, OnDestroy {
+    private dragStartSubscription: Subscription;
+    private dragMoveSubscription: Subscription;
+    private dragEndSubscription: Subscription;
+
+    private droppableEl: Node;
+    private clientRect: ClientRect;
+
+    constructor(private el: ElementRef, private eventBus: ClrDragAndDropEventBus<T>, private domAdapter: DomAdapter,
+                private renderer: Renderer2) {
+        this.droppableEl = this.el.nativeElement;
+    }
+
+    private isDraggableMatch: boolean = false;
+    private _isDraggableOver: boolean = false;
+
+    set isDraggableOver(value: boolean) {
+        if (value) {
+            this.renderer.addClass(this.droppableEl, "draggable-over");
+        } else {
+            this.renderer.removeClass(this.droppableEl, "draggable-over");
+        }
+        this._isDraggableOver = value;
+    }
+
+    private _group: string|string[];
+    private _dropTolerance: ClrDropTolerance = {top: 0, right: 0, bottom: 0, left: 0};
+
+    @Input("clrGroup")
+    set group(value: string|string[]) {
+        this._group = value;
+    }
+
+    @Input("clrDropTolerance")
+    set dropTolerance(value: number|ClrDropTolerance) {
+        if (typeof value === "number") {
+            this._dropTolerance.top = value;
+            this._dropTolerance.right = value;
+            this._dropTolerance.bottom = value;
+            this._dropTolerance.left = value;
+        } else if (value) {
+            if (value.top) {
+                this._dropTolerance.top = value.top;
+            }
+            if (value.right) {
+                this._dropTolerance.right = value.right;
+            }
+            if (value.bottom) {
+                this._dropTolerance.bottom = value.bottom;
+            }
+            if (value.left) {
+                this._dropTolerance.left = value.left;
+            }
+        }
+    }
+
+    @Output("clrDragStart") dragStartEmitter: EventEmitter<ClrDragEvent<T>> = new EventEmitter();
+    @Output("clrDragMove") dragMoveEmitter: EventEmitter<ClrDragEvent<T>> = new EventEmitter();
+    @Output("clrDragEnd") dragEndEmitter: EventEmitter<ClrDragEvent<T>> = new EventEmitter();
+    @Output("clrDragLeave") dragLeaveEmitter: EventEmitter<ClrDragEvent<T>> = new EventEmitter();
+    @Output("clrDragEnter") dragEnterEmitter: EventEmitter<ClrDragEvent<T>> = new EventEmitter();
+    @Output("clrDrop") dropEmitter: EventEmitter<ClrDragEvent<T>> = new EventEmitter();
+
+    private unsubscribeFrom(subscription: Subscription): void {
+        if (subscription) {
+            subscription.unsubscribe();
+        }
+    }
+
+    private checkGroupMatch(draggableGroup: string|string[]): boolean {
```

Comment:
This logic is not too intuitive to me, could you annotate a little bit about these conditionals?

Reply:
I have to turn off this specific subscription in multiple different cases/places. That's why I haven't pushed it into an array of subscriptions.